internal/wasm/module.go

```diff
@@ -555,7 +555,7 @@ func (m *Module) buildGlobals(importedGlobals []*GlobalInstance) (globals []*Glo
 		g := &GlobalInstance{Type: gs.Type}
 		switch v := executeConstExpression(importedGlobals, gs.Init).(type) {
 		case int32:
-			g.Val = uint64(v)
+			g.Val = uint64(uint32(v))
```

Comment:
previously this was sign-extending the i32 integers so higher bits were unnecessarily set for i32 globals.